src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.Dgml.cs

```diff
@@ -0,0 +1,212 @@
+            writer.WriteLine("            <Setter Property=\"FontFamily\" Value=\"Arial\" />");
+            writer.WriteLine("        </Style>");
+            writer.WriteLine("        <Style TargetType=\"Link\" GroupLabel=\"StartTransition\" ValueLabel=\"True\">");
+            writer.WriteLine("            <Condition Expression=\"HasCategory('StartTransition')\" />");
+            writer.WriteLine("        </Style>");
+            writer.WriteLine("        <Style TargetType=\"Link\" GroupLabel=\"EpsilonTransition\" ValueLabel=\"True\">");
+            writer.WriteLine("            <Condition Expression=\"HasCategory('EpsilonTransition')\" />");
+            writer.WriteLine("            <Setter Property=\"StrokeDashArray\" Value=\"8 8\" />");
+            writer.WriteLine("        </Style>");
+            writer.WriteLine("        <Style TargetType=\"Link\" GroupLabel=\"FinalLabel\" ValueLabel=\"False\">");
+            writer.WriteLine("            <Condition Expression=\"HasCategory('FinalLabel')\" />");
+            writer.WriteLine("            <Setter Property=\"StrokeDashArray\" Value=\"8 8\" />");
+            writer.WriteLine("        </Style>");
+            writer.WriteLine("        <Style TargetType=\"Node\" GroupLabel=\"StateInfo\" ValueLabel=\"True\">");
+            writer.WriteLine("            <Setter Property=\"Stroke\" Value=\"white\" />");
+            writer.WriteLine("            <Setter Property=\"FontSize\" Value=\"18\" />");
+            writer.WriteLine("            <Setter Property=\"FontFamily\" Value=\"Arial\" />");
+            writer.WriteLine("        </Style>");
+            writer.WriteLine("        <Style TargetType=\"Node\" GroupLabel=\"DFAInfo\" ValueLabel=\"True\">");
+            writer.WriteLine("            <Setter Property=\"Stroke\" Value=\"white\" />");
+            writer.WriteLine("            <Setter Property=\"FontSize\" Value=\"18\" />");
+            writer.WriteLine("            <Setter Property=\"FontFamily\" Value=\"Arial\" />");
+            writer.WriteLine("        </Style>");
+            writer.WriteLine("    </Styles>");
+            writer.WriteLine("</DirectedGraph>");
+
+            // This function gathers all transitions in the given builder and groups them by (source,destination) state ID.
+            // It also counts the number of transitions found.
+            static Dictionary<(int, int), (TSet, List<int>)> GatherTransitions(SymbolicRegexBuilder<TSet> builder)
+            {
+                Debug.Assert(builder._delta is not null);
+                Debug.Assert(builder._minterms is not null);
+                Dictionary<(int, int), (TSet, List<int>)> result = new();
+                foreach (DfaMatchingState<TSet> source in builder._stateCache)
+                {
+                    // Get the span of entries in delta that gives the transitions for the different minterms
+                    Span<DfaMatchingState<TSet>?> deltas = builder.GetDeltasSpanFor(source);
```

Comment:
Do we need "Span" in the name of these helpers, or could they just be GetDfaDeltasFor and GetNfaDeltasFor?

Reply:
I added the "Span" to clarify that they just give sections of the deltas tables including null elements, rather than gathering the current valid ones oslt, but the type does indicate that too.